src/main/java/com/impactupgrade/nucleus/service/segment/StripeDataSyncService.java

```diff
@@ -31,18 +42,85 @@ public void init(Environment env) {
   public void syncContacts(Calendar updatedAfter) throws Exception {
     PagedResults<CrmContact> contactPagedResults = env.primaryCrmService().getDonorIndividualContacts(updatedAfter);
     for (PagedResults.ResultSet<CrmContact> resultSet : contactPagedResults.getResultSets()) {
-      //TODO: bulk update?
       for (CrmContact crmContact : resultSet.getRecords()) {
-        //TODO: update contact in Stripe
+        try {
+          updateOrCreateCustomer(crmContact);
+        } catch (Exception e) {
+          env.logJobError("{}/syncContacts failed: {}", this.name(), e);
+        }
       }
     }
   }
 
   @Override
   public void syncTransactions(Calendar updatedAfter) throws Exception {
-    List<CrmDonation> crmDonations = env.primaryCrmService().getDonations(updatedAfter);
-    for (CrmDonation crmDonation : crmDonations) {
-      //TODO: bulk update in Stripe
+    //TODO: pull our nightly batch job from PaymentGatewayController
+    // and instead have this concept do it
+  }
+
+  protected Customer updateOrCreateCustomer(CrmContact crmContact) throws Exception {
+    Optional<Customer> existingCustomer = getCustomer(crmContact);
+    if (existingCustomer.isPresent()) {
+      // Update
+      CustomerUpdateParams customerUpdateParams = CustomerUpdateParams.builder()
+          .setName(crmContact.getFullName())
+          .setEmail(crmContact.email.toLowerCase(Locale.ROOT))
+          .setAddress(toUpdateAddress(crmContact.account.billingAddress))
+          .setPhone(crmContact.mobilePhone)
+          .build();
+      return callWithRetries(() -> env.stripeClient().updateCustomer(existingCustomer.get(), customerUpdateParams));
+    } else {
+      //TODO: create?
```

Comment:
Nope, kill this part -- only updates. SFDC never creates Stripe Customers. Their donation does that.